cmd/release/main.go

```diff
@@ -0,0 +1,164 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+)
+
+/*
+required parameters
+
+https://help.github.com/en/articles/creating-a-personal-access-token-for-the-command-line
+github token for modifying release
+
+# dependencies
+* requires a build environment (make artifact-release must work)
+* github.com/itchio/gothub
+*/
+
+const (
+	// TODO(chuckha): figure this out based on directory name
+	repository   = "cluster-api-provider-aws"
+	artifactsDir = "out"
+)
+
+func main() {
+	if len(os.Args) == 1 || os.Args[1] == "help" {
+		documentation()
+		return
+	}
+
+	mustSatisfyDependencies()
+
+	// TODO(chuckha) consider moving all these things out to some config struct so other
+	// providers can reuse this release
+	expectedFiles := []string{
+		// TODO(chuckha) these aren't examples. rename them.
+		"cluster-api-provider-aws-examples.tar",
+		"clusterawsadm-darwin-amd64",
+		"clusterawsadm-linux-amd64",
+		"clusterctl-darwin-amd64",
+		"clusterctl-linux-amd64",
+	}
+
+	// TODO(chuckha): it would be ideal if we could release major/minor/patch and have it
+	// automatically bump the latest tag git finds
+	// until then, hard code it
+	remote := os.Args[1] // default to origin probably
+	user := os.Args[2]   // default to kubernetes-sigs
+	version := os.Args[3]
+
+	// Build all the release binaries
+	RunCommand("make", "release-artifacts")
+
```

Comment:
suggest adding a validate artifacts step to confirm all expected files are present.